src/DynamoCoreWpf/ViewModels/Core/PortViewModel.cs

```diff
@@ -511,49 +512,120 @@ private void AutoCompleteCluster(object parameter)
+                            // This is the target node from cluster that should connect to the query node
+                            targetNodeFromCluster = nodeFromCluster;
+                        }
+                        index++;
+                    });
+
+                    ClusterResultItem.Topology.Connections.ToList().ForEach(connection =>
+                    {
+                        // Connect the nodes
+                        var sourceNode = clusterMapping[connection.StartNode.NodeId];
+                        var targetNode = clusterMapping[connection.EndNode.NodeId];
+                        // The port index is 1- based (currently a hack and not expected from service)
+                        var sourcePort = sourceNode.OutPorts.FirstOrDefault(p => p.PortModel.Index == connection.StartNode.PortIndex - 1);
+                        var targetPort = targetNode.InPorts.FirstOrDefault(p => p.PortModel.Index == connection.EndNode.PortIndex - 1);
+                        var commands = new List<DynamoModel.ModelBasedRecordableCommand>
+                        {
+                            new DynamoModel.MakeConnectionCommand(sourceNode.Id.ToString(), connection.StartNode.PortIndex - 1, PortType.Output, DynamoModel.MakeConnectionCommand.Mode.Begin),
+                            new DynamoModel.MakeConnectionCommand(targetNode.Id.ToString(), connection.EndNode.PortIndex - 1, PortType.Input, DynamoModel.MakeConnectionCommand.Mode.End),
+                        };
+                        commands.ForEach(c =>
+                        {
+                            try
+                            {
+                                wsViewModel.DynamoViewModel.Model.ExecuteCommand(c);
+                            }
+                            catch (Exception) { }
+                        });
+                    });
+
+                    // Connect the cluster to the original node and port
+                    var finalCommands = new List<DynamoModel.ModelBasedRecordableCommand>
+                    {
+                        new DynamoModel.MakeConnectionCommand(node.Id.ToString(), 0, PortType.Output, DynamoModel.MakeConnectionCommand.Mode.Begin),
+                        new DynamoModel.MakeConnectionCommand(targetNodeFromCluster?.Id.ToString(), ClusterResultItem.EntryNodeInPort, PortType.Input, DynamoModel.MakeConnectionCommand.Mode.End),
+                    };
+                    finalCommands.ForEach(c =>
+                    {
+                        try
+                        {
+                            wsViewModel.DynamoViewModel.Model.ExecuteCommand(c);
+                        }
+                        catch (Exception) { }
+                    });
+
+                    // AutoLayout should be called after all nodes are connected
+                    foreach(var node in clusterMapping.Values)
+                    {
+                        wsViewModel.DynamoViewModel.Model.AddToSelection(node.NodeModel);
```

Comment:
A nice benefit here is that you can move them all together after the placement since they are still in the selection. Perhaps we don't need a group and this will suffice I think ?

Reply:
yeah I find it useful for now, I have not consulted @Jingyi-Wen yet if we want to apply temp group and transiant state, etc. We can ask later, for the moment this should be good